### PR TITLE
watermark added

### PR DESCRIPTION
**Pull request by cihadtekinbas**

### CODE REVIEW

**Comment by cihadtekinbas** [comment]
pip install scipy need

File: backend/watermark/helpers.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                          AutoModelForCausalLM,
+                          LogitsProcessorList)
+import torch
+import argparse
+from argparse import Namespace
+
+arg_dict = {
+    'run_gradio': True, 
+    'demo_public': False, 
+     'model_name_or_path': 'facebook/opt-125m',
```
Comment (cihadtekinbas) [inline backend/watermark/helpers.py:16]: if it is possible can we test with large models? @muratbayrktr

File: backend/watermark/utils.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    output_tokens = model.generate(**tokenized_input,
+                                logits_processor=LogitsProcessorList([watermark_processor]))
+
+    # if decoder only model, then we need to isolate the
+    # newly generated tokens as only those are watermarked, the input/prompt is not
+    output_tokens = output_tokens[:,tokenized_input["input_ids"].shape[-1]:]
+
+    output_text = tokenizer.batch_decode(output_tokens, skip_special_tokens=True)[0]
+    print(f"watermarked:{output_text}")
+    return output_text
```
Comment (cihadtekinbas) [inline backend/watermark/utils.py:75]: can we add watermarked text box in the view? if the model selected watermark @muratbayrktr

Comment (muratbayrktr) [reply backend/watermark/utils.py:75]: Yes, we can I will implement it at my soonest convenience

File: backend/watermark/helpers.py (hunk 1 above)
Comment (muratbayrktr) [reply backend/watermark/helpers.py:16]: I believe we can, what is the size of the `large models` you refer?

**Review by muratbayrktr — APPROVED** [review]

File: backend/watermark/helpers.py (hunk 1 above)
Comment (cihadtekinbas) [reply backend/watermark/helpers.py:16]: #'model_name_or_path': 'facebook/opt-6.7b',
    # 'model_name_or_path': 'facebook/opt-13b',

**Pull request merged**